### DISCUSSION TITLE
C++ Development Practice

### DISCUSSION THREAD

**Discussion by TLWise** [idaholab/moose#27074] [Q&A Getting Started] [answered]
In an effort to increase my ability to comprehend and apply C++ programming to my MOOSE applications, I would like to make and test the output for coding such as "Hello World". Is it possible to make and test simple programs, not associated with numerical analysis in the MOOSE environment, or do I need to install a dedicated C++ compiler?

**Comment by GiudGiud** [comment, marked as answer]
Hello

You can use the compilers we shipped with the conda packages to compile simple programs as well.
There's a hello world script here https://mooseframework.inl.gov/help/faq/build_issues.html
you can use

Guillaume

**Reply by TLWise** [reply]
Great! Thank you for the verification and the provided link.
